WordPress/Classes/Utility/Environment/Environment.swift

```diff
@@ -0,0 +1,53 @@
+
+import Foundation
+
+/// A collection of global variables and singletons that the app wants access to.
+///
+struct Environment {
+
+    // MARK: - Globals
+
+    /// A type that helps tracking whether or not a user should be prompted for an app review
+    let appRatingUtility: AppRatingUtilityType
+
+    /// A type to create derived context, save context, etc...
+    let contextManagerType: ContextManagerType.Type
+
+    /// The mainContext that has concurrency type NSMainQueueConcurrencyType and should be used
+    /// for UI elements and fetched results controllers.
+    var mainContext: NSManagedObjectContext {
+        return contextManagerType.shared.mainContext
+    }
+
+    // MARK: - Static current environment implementation
+
+    /// The current environment. Use this to access the app globals.
+    ///
+    static private(set) var current = Environment()
+
+    // MARK: - Initialization
+
+    private init(
+        appRatingUtility: AppRatingUtilityType = AppRatingUtility.shared,
+        contextManagerType: ContextManagerType.Type = ContextManager.self) {
```

Comment:
I'm a little confused – why don't we handle context manager the same way as `appRatingUtility`? So `shared` isn't part of the protocol, and we just have `ContextManager.shared` as the default value here.

Reply:
You are completely right! I got confused. 😅
This is useful for when the class has static methods and this is not the case. I'll fix it right now.